src/circuit/gadget/ecc/chip/mul_fixed.rs

```diff
@@ -0,0 +1,529 @@
+        for window in 0..NUM_WINDOWS {
+            // Enable `q_mul_fixed` selector
+            self.q_mul_fixed.enable(region, window + offset)?;
+
+            // Assign x-coordinate Lagrange interpolation coefficients
+            for k in 0..(constants::H) {
+                region.assign_fixed(
+                    || {
+                        format!(
+                            "Lagrange interpolation coeff for window: {:?}, k: {:?}",
+                            window, k
+                        )
+                    },
+                    self.lagrange_coeffs[k],
+                    window + offset,
+                    || Ok(lagrange_coeffs[window].0[k]),
+                )?;
+            }
+
+            // Assign z-values for each window
+            region.assign_fixed(
+                || format!("z-value for window: {:?}", window),
+                self.fixed_z,
+                window + offset,
+                || Ok(z[window]),
+            )?;
+        }
+
+        Ok(())
+    }
+
+    fn copy_scalar(
+        &self,
+        region: &mut Region<'_, pallas::Base>,
+        offset: usize,
+        scalar: &ScalarFixed,
+    ) -> Result<(), Error> {
+        // Copy the scalar decomposition (`k`-bit windows)
+        for (window_idx, window) in scalar.windows().iter().enumerate() {
+            copy(
+                region,
+                || format!("k[{:?}]", window),
+                self.window,
+                window_idx + offset,
+                window,
+                &self.perm,
+            )?;
+        }
+
+        Ok(())
+    }
+
+    fn initialize_accumulator(
+        &self,
+        region: &mut Region<'_, pallas::Base>,
+        offset: usize,
+        base: OrchardFixedBases,
+        scalar: &ScalarFixed,
+    ) -> Result<EccPoint, Error> {
+        // Recall that the message at each window `w` is represented as
+        // `m_w = [(k_w + 2) ⋅ 8^w]B`.
+        // When `w = 0`, we have `m_0 = [(k_0 + 2)]B`.
+        let m0 = {
+            let k0 = scalar.windows_field()[0];
+            let m0 = k0.map(|k0| base.generator() * (k0 + pallas::Scalar::from_u64(2)));
+            let m0 = m0.map(|m0| m0.to_affine().coordinates().unwrap());
+
+            let x = m0.map(|m0| *m0.x());
+            let x_cell = region.assign_advice(
+                || "m0_x",
+                self.x_p,
+                offset,
+                || x.ok_or(Error::SynthesisError),
+            )?;
+            let x = CellValue::new(x_cell, x);
+
+            let y = m0.map(|m0| *m0.y());
+            let y_cell = region.assign_advice(
+                || "m0_y",
+                self.y_p,
+                offset,
+                || y.ok_or(Error::SynthesisError),
+            )?;
+            let y = CellValue::new(y_cell, y);
+
+            EccPoint { x, y }
+        };
+
+        // Assign u = (y_p + z_w).sqrt() for `m0`
+        {
+            let k0 = scalar.windows_usize()[0];
+            let u0 = &base.u()[0];
```

Comment:
Same thing happens twice here (inside `base.generator()` and `base.u()`):
![image](https://user-images.githubusercontent.com/4993799/122225392-efee4a00-ceac-11eb-9f49-29e6e557980f.png)